includes/payment-methods/class-wc-stripe-payment-request.php

```diff
@@ -763,9 +765,26 @@ public function ajax_get_shipping_options() {
 				throw new Exception( __( 'Unable to find shipping method for address.', 'woocommerce-gateway-stripe' ) );
 			}
 
-			if ( isset( $data[0] ) ) {
+			// Check if at least single shipping option is available for new address.
+			if ( isset( $data['shipping_options'][0] ) ) {
+				if ( isset( $chosen_shipping_methods[0] ) ) {
+					$chosen_method_id         = $chosen_shipping_methods[0];
+					$compare_shipping_options = function ( $a, $b ) use ( $chosen_method_id ) {
```

Comment:
Why not simply use the `$chosen_method_id` and pass it to `update_shipping_method` as a wrapped array?  If we do that would we still need to do a sort? The sort seems like it is not used beyond simply making sure the selected is sorted to be the first item and then using that.

Reply:
> Why not simply use the $chosen_method_id and pass it to update_shipping_method as a wrapped array?

$chosen_method_id might be not available for new address.

> If we do that would we still need to do a sort? The sort seems like it is not used beyond simply making sure the selected is sorted to be the first item and then using that.

We do sorting to make sure the chosen method is the first in the list or the list should remain unchanged. So if chosen method still available we'll apply it otherwise we fall back to the first in the list.

Reply:
Thanks for explaining and adding the comment.